libs/clients/smartsolutions/src/lib/smartSolutions.api.ts

```diff
@@ -210,49 +210,92 @@ export class SmartSolutionsApi {
       data: undefined,
     }
   }
-
-  async verifyPkPass(
-    payload: DynamicBarcodeDataInput,
-  ): Promise<Result<VerifyPassData>> {
+  /**
+   *
+   * @param payload What the created/updated pass should contain
+   * @returns The created/updated pass
+   */
+  private async postPass(payload: PassDataInput): Promise<Result<Pass>> {
     const graphql = JSON.stringify({
-      query: VERIFY_PKPASS,
+      query: UPSERT_PASS,
       variables: {
-        dynamicBarcodeData: payload,
+        inputData: {
+          passTemplateId: this.config.passTemplateId,
+          ...payload,
+        },
       },
     })
 
-    const res = await this.query<VerifyPassResponseData>(graphql)
+    const res = await this.query<UpsertPassResponseData>(graphql)
 
     if (res.ok) {
       return {
         ok: true,
-        data: {
-          valid: true,
-          pass: res.data.updateStatusOnPassWithDynamicBarcode,
-        },
+        data: res.data.upsertPass,
       }
     }
 
     return res
   }
 
-  async upsertPkPass(payload: PassDataInput): Promise<Result<Pass>> {
+  private async updateExistingPass(
+    payload: PassDataInput,
+    passId: string,
+  ): Promise<Result<Pass>> {
+    const getPassQuery = JSON.stringify({
+      query: GET_PASS,
+      variables: {
+        id: passId,
+      },
+    })
+
+    const getPassRes = await this.query<GetPassResponseData>(getPassQuery)
+
+    if (!getPassRes.ok) {
+      //if failure, return the response
+      return getPassRes
+    }
+    const passInputData = mapPassToPassDataInput(getPassRes.data.pass)
+
+    const inputFieldValues = mergeInputFields(
+      passInputData.inputFieldValues ?? undefined,
+      payload.inputFieldValues ?? undefined,
+    )
+    //now we finally have the updated pass data!
```

Comment:
```suggestion
```

it's an easy process now, no need to be dramatic 😎

Reply:
It wasn't until it was!